crates/sqlexec/src/planner/errors.rs

```diff
@@ -52,6 +52,11 @@ pub enum PlanError {
     #[error("Invalid alter statement: {msg}")]
     InvalidAlterStatement { msg: &'static str },
 
+    #[error("Invalid copy to statement: {source}")]
+    InvalidCopyToStatement {
+        source: Box<dyn std::error::Error + Send + Sync>,
```

Comment:
`ValidationError::CredentialsNotSupportedByDatasource`

this error is always going to a Validation error, seems like we don't need the generic here?

Reply:
No, not really. Right now it is. But this is a flexible error to mark any invalid copy to statement. Something like `InvalidCopyToStatement(ValidationError)` implies that you can only have an invalid copy to statement due to a validation of options issue.